java-components/cli/src/main/java/io/github/redhatappstudio/jvmbuild/cli/builds/BuildDiagnosticCommand.java

```diff
@@ -0,0 +1,122 @@
+                    if (build.equals(n.getMetadata().getName())) {
+                        //can also specify by kube name
+                        theBuild = n;
+                        break;
+                    }
+                }
+            }
+        } else if (artifact != null) {
+            if (gav != null) {
+                throwUnspecified();
+            }
+            ArtifactBuild ab = ArtifactBuildCompleter.createNames().get(artifact);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else if (gav != null) {
+            ArtifactBuild ab = GavCompleter.createNames().get(gav);
+            theBuild = BuildConverter.buildToArtifact(client, ab);
+        } else {
+            throw new RuntimeException("Must specify one of -b, -a or -g");
+        }
+        if (theBuild == null) {
+            throw new RuntimeException("Build not found");
+        }
+        int lastSlash = theBuild.getSpec().getScm().getScmURL().lastIndexOf('/') + 1;
+        String name = theBuild.getSpec().getScm().getScmURL().substring(lastSlash).replaceFirst("\\.git.*", "");
+        System.out.println("Target directory: " + targetDirectory);
+        System.out.println(
+                "Selected build: " + theBuild.getMetadata().getName() + " of " + theBuild.getSpec().getScm().getScmURL() + ':' +
+                        theBuild.getSpec().getVersion());
+
+        List<String> dockerFiles = theBuild.getStatus().getDiagnosticDockerFiles();
+        if ((theBuild.getStatus().getFailedBuildRecipes() == null ? 0 : theBuild.getStatus().getFailedBuildRecipes().size())
+                + 1 != dockerFiles.size()) {
+            throw new RuntimeException("Mismatch between failed/current build recipe count and number of Dockerfiles.");
+        }
+        try {
+            for (int i = 0; i < dockerFiles.size(); i++) {
+                String fileName;
+                if (i == dockerFiles.size() - 1) {
+                    // Final one - which is successful.
+                    fileName = "Dockerfile." + name + ".succeed.jdk" + theBuild.getStatus()
+                            .getCurrentBuildRecipe()
+                            .getJavaVersion();
+                } else {
+                    fileName = "Dockerfile." + name + ".failed.jdk" + theBuild.getStatus()
+                            .getFailedBuildRecipes()
+                            .get(i)
+                            .getJavaVersion();
+                }
+                System.out.println("Writing " + fileName);
+                Files.writeString(Paths.get(targetDirectory.toString(), fileName),
+                        dockerFiles.get(i));
+            }
+            System.out.println(CommandLine.Help.Ansi.AUTO
+                    .string("""
+                            \n@|green To build the image, a standard build command e.g. |@@|yellow podman build . |@@|green (or |@@|yellow podman build -f <Dockerfile> .|@@|green ) is sufficient.
```

Comment:
Minor nit, but can we list the commands for the specific image generated?

e.g. Something like


```
To diagnose the JDK8 Build:

podman build . -f Dockerfile.somebuild.failed.jdk8 -t somebuild.jdk8
podman run -it somebuild.jdk8

To diagnose the JDK11 Build:

podman build . -f Dockerfile.somebuild.failed.jdk11 -t somebuild.jdk11
podman run -it somebuild.jdk11
```

Reply:
Done. Typical output now looks like:

```
java -jar target/quarkus-app/quarkus-run.jar build diagnostic -b=0aeae1d9bc1758ff9e368b527280024b
Target directory: /home/rnc/Work/Apheleia/jvm-build-service/java-components/cli
Selected build: 0aeae1d9bc1758ff9e368b527280024b of https://github.com/jvm-build-service-code/wsdl4j.git:1.6.3

To diagnose the JDK8 build:

podman build -f Dockerfile.wsdl4j.succeed.jdk8 -t wsdl4j.succeed.jdk8 .
podman run -it wsdl4j.succeed.jdk8
```